stat_fem/InterpolationMatrix.py

```diff
@@ -0,0 +1,188 @@
+import sys
+import numpy as np
+from firedrake import Function
+from firedrake.petsc import PETSc
+from firedrake.functionspace import VectorFunctionSpace
+from firedrake.functionspaceimpl import WithGeometry
+from firedrake.interpolation import interpolate
+from firedrake.matrix import Matrix
+from firedrake.vector import Vector
+
+class InterpolationMatrix(object):
+    "class representing an interpolation matrix"
+    def __init__(self, function_space, coords):
+        "create and assemble interpolation matrix"
+
+        if not isinstance(function_space, WithGeometry):
```

Comment:
You are correct that that is the "only do anything once" solution to having this in every class. I'm not sure it's worth the overhead here if that's the only thing that is checked, because you will still need a line of code to call the superclass's init method. However, there might be some more things that are shared (particularly between the dressed up sparse matrix classes), so I will look into that.

Reply:
Raised issue #28.